pkg/controller/nodeipam/ipam/multicidrset/multi_cidr_set_test.go

```diff
@@ -801,24 +859,67 @@ func TestMultiCIDRSetMetricsDual(t *testing.T) {
 		allocs:     65536,
 		releases:   65536,
 		allocTries: 0,
+		max:        float64(maxIPv6),
 	}
 	expectMetrics(t, cidrIPv6, em)
 
 }
 
+func Test_getMaxCIDRs(t *testing.T) {
+	cidrIPv4 := "10.0.0.0/16"
+	_, clusterCIDRv4, _ := utilnet.ParseCIDRSloppy(cidrIPv4)
+
+	cidrIPv6 := "2001:db8::/48"
+	_, clusterCIDRv6, _ := utilnet.ParseCIDRSloppy(cidrIPv6)
+
+	tests := []struct {
+		name             string
+		subNetMaskSize   int
+		clusterCIDR      *net.IPNet
+		expectedMaxCIDRs int
+	}{
+		{
+			name:             "IPv4",
+			subNetMaskSize:   24,
+			clusterCIDR:      clusterCIDRv4,
+			expectedMaxCIDRs: 256,
+		},
+		{
+			name:             "IPv6",
+			subNetMaskSize:   64,
+			clusterCIDR:      clusterCIDRv6,
+			expectedMaxCIDRs: 65536,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			clusterMaskSize, _ := test.clusterCIDR.Mask.Size()
+			maxCIDRs := getMaxCIDRs(test.subNetMaskSize, clusterMaskSize)
+			if test.expectedMaxCIDRs != maxCIDRs {
+				t.Errorf("incorrect maxCIDRs, expected: %d, got: %d", test.expectedMaxCIDRs, maxCIDRs)
+			}
+		})
+	}
+}
+
 // Metrics helpers.
-func clearMetrics(labels map[string]string) {
+func clearMetrics(labels map[string]string, clearMax bool) {
```

Comment:
why the bool?

Reply:
it's a flag to indicate whether we want to clear the `cidrSetMaxMetrics` as well. this is helpful in tests, since `cidrSetMaxMetrics` is set at the very beginning during `NewMultiCIDRSet` and not touched again in calls to `Release` and `Occupy`.

Reply:
but it is always true, why do we want that if we never use it?